### PR TITLE
Pass type arguments through dots

### PR DESCRIPTION
**Pull request by zeileis**
Today I took a quick stab at passing arguments through the dots `...` to the `type_*` function. The main motivation for this came from https://github.com/grantmcdermott/tinyplot/pull/233#issuecomment-2443010778. Because:

```
plot(Species ~ Sepal.Width, data = iris, breaks = 4)
```

works, I also wanted to get

```
tinyplot(Species ~ Sepal.Width, data = iris, breaks = 4)
```

to work rather than having to call

```
tinyplot(Species ~ Sepal.Width, data = iris, type = type_spineplot(breaks = 4))
```

Similarly, instead of

```
plot(count ~ spray, data = InsectSprays, lty = 1, staplewex = 0)
```

we can now also do

```
tinyplot(count ~ spray, data = InsectSprays, lty = 1, staplewex = 0)
```

rather than having to use

```
tinyplot(count ~ spray, data = InsectSprays, lty = 1, type = type_boxplot(staplewex = 0))
```

tinyplot-dots

I've implemented this by:

- Passing the `dots` also to `sanitize_type`.
- Obtaining the `type` function without evaluating it yet.
- Taking those named arguments from `dots` that match arguments of `type` using them to evaluate `type`.
- Returning the remainings `dots` back to the `tinyplot`.

Thus, this only works for `type` arguments that are not arguments of `tinyplot` \_and\_ do not even partially match! So this is a potential source of confusion. But it does avoid the potential confusion from examples like the ones above.

### CODE REVIEW

**Comment by vincentarelbundock** [comment]
This seems pretty straightforward and useful. 

We may still want to strictly advertise the "old" system on the website and in the docs, because it clearly indicates where users should look for to get documentation on type-specific arguments. 

Perhaps showing a call-out box with a "tip" about `...`, while staying ultra consistent throughout the website. 

Not quite the "there should be only one way" Tao, but...

**Comment by zeileis** [comment]
I've now also added `"function"` to the official `types` in `sanitize_type()` now. It does not strictly belong to this PR but I thought that it fits sufficiently well and it's only a minor edit.

**Comment by vincentarelbundock** [comment]
I've looked at the code and don't see any problem. Do you plan to do more work on this? If not, I think we should just merge it.

**Comment by zeileis** [comment]
Great, thanks for the feedback, Vincent. I also don't think that further work is needed. So let's see whether Grant @grantmcdermott has any concerns...

**Review by grantmcdermott — CHANGES_REQUESTED** [review]
Thanks Achim. I agree that this is helpful and adds a disciplined "backend" for users that would like to pass customization arguments without having to go through the `type_*()` functions.

On that note, we'll also need to edit the first "breaking changes" item in NEWS here. https://github.com/grantmcdermott/tinyplot/blob/d76f4f1cd5adc1155a1160eb6a3dc815dc3245e8/NEWS.md?plain=1#L9-L25
Do you mind taking a crack at the re-wording? We can obviously move it out of breaking changes, since the old way will work again with your PR. But we should underscore that the documentation lives with the individual type functions and politely suggest that passing arguments through them (rather than `...`) is the more idiomatic tinyplot way of doing things now.

File: R/sanitize.R [hunk 1]
```diff
@@ -15,52 +15,59 @@ sanitize_type = function(type, x, y) {
...
+    "j" = type_jitter,
+    "jitter" = type_jitter,
+    "loess" = type_loess,                                    
+    "ridge" = type_ridge,
+    "spline" = type_spline,
+    "glm" = type_glm,
+    "lm" = type_lm,
+    "function" = type_function,
+    type # Default case
   )
```
[the remark this answers is not in the document]
Comment (zeileis) [reply R/sanitize.R:61]: I have rephrased/reorganized the NEWS. Please have a look at whether you like this approach or would prefer to do it differently. Maybe there should also be something in the vignette?

I also noticed that there is still a mentioning of `ribbon.apha` in `tinyplot.Rd` which can probably go.

I fully agree with the other proposed changes and will do them a bit later...

[the remark this answers is not in the document]
Comment (grantmcdermott) [reply R/sanitize.R:61]: > I also noticed that there is still a mentioning of `ribbon.apha` in `tinyplot.Rd` which can probably go.

Good catch! Yes, please remove.

[the remark this answers is not in the document]
Comment (zeileis) [reply R/sanitize.R]: All done. Please check the NEWS update to see whether you like it - and whether you agree that the non-standard spacing is helpful in https://github.com/grantmcdermott/tinyplot/blob/type-dots/R/sanitize.R#L36-L64

Open: Add something in vignette? Add tests, e.g., checking that `tinyplot(Nile, type = "hist", breaks  30)` is indeed identical to `tinyplot(Nile, type = type_hist(breaks = 30))`?

**Comment by grantmcdermott** [comment]
Great stuff, thanks @zeileis!

**Pull request merged**